File: EvolutionaryGenerator_v2.1_Python/metah/generator.py
"""Translation of ``mx.tec.meta.Generator``."""

from abc import ABC, abstractmethod

from metah.individual import Individual


class Generator(ABC):
    """Defines the methods that need to be implemented by a generator to be used by the meta heuristics.

    :author: Paola Azeneth Castillo Gutiérrez (Python translation)
    :original author: Jose Carlos Ortiz Bayliss (jcobayliss@tec.mx)
    :original version: 2.0
    """

    @abstractmethod
    def generate(self) -> Individual:
        """Generates a new random instance of ``Solution``.

        :return: A random solution.
        """

File: EvolutionaryGenerator_v2.1_Java/MetaH/src/mx/tec/meta/Generator.java
package mx.tec.meta;

/**
 * Defines the methods that need to be implemented by a generator to be used by the meta heuristics.
 *
 * @author Jose Carlos Ortiz Bayliss (jcobayliss@tec.mx)
 * @version 2.0
 */
public abstract class Generator {

    /**
     * Generates a new random instance of <code>Solution</code>.
     *
     * @return A random solution.
     */
    public abstract Individual generate();

}
